From 88b4df83a567d4537de622a33c5026a263b71f6b Mon Sep 17 00:00:00 2001
From: De Huo <De.Huo@windriver.com>
Date: Thu, 15 Aug 2019 14:25:37 +0800
Subject: [PATCH] Rule sysctl_net_ipv4_conf_all_send_redirects add bash
 script

Rule sysctl_net_ipv4_conf_all_send_redirects add oval check for WRLinux1019.

Upstream-Status: Pending

Signed-off-by: De Huo <De.Huo@windriver.com>
---
 .../bash/wrlinux1019.sh                              | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)
 create mode 100644 linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_net_ipv4_conf_all_send_redirects/bash/wrlinux1019.sh

diff --git a/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_net_ipv4_conf_all_send_redirects/bash/wrlinux1019.sh b/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_net_ipv4_conf_all_send_redirects/bash/wrlinux1019.sh
new file mode 100644
index 0000000..b83f642
--- /dev/null
+++ b/linux_os/guide/system/network/network-kernel/network_host_parameters/sysctl_net_ipv4_conf_all_send_redirects/bash/wrlinux1019.sh
@@ -0,0 +1,20 @@
+# platform = multi_platform_wrlinux
+
+. /usr/share/scap-security-guide/remediation_functions
+
+#
+# Set runtime for net.ipv4.conf.all.send_redirects
+#
+/sbin/sysctl -q -n -w net.ipv4.conf.all.send_redirects=0
+
+#
+# If net.ipv4.conf.all.send_redirects in /etc/sysctl.conf, change value to appropriate value
+#       else, add "net.ipv4.conf.all.send_redirects = 0" to /etc/sysctl.conf
+#
+if grep --silent ^net.ipv4.conf.all.send_redirects /etc/sysctl.conf ; then
+        sed -i "s/^net.ipv4.conf.all.send_redirects.*/net.ipv4.conf.all.send_redirects = 0/g" /etc/sysctl.conf
+else
+        echo -e "\n# Set net.ipv4.conf.all.send_redirects to 0 per security requirements" >> /etc/sysctl.conf
+        echo "net.ipv4.conf.all.send_redirects = 0" >> /etc/sysctl.conf
+fi
+
-- 
1.9.1

